packages/flutter_goldens/lib/flutter_goldens.dart

```diff
@@ -93,14 +164,134 @@ class FlutterGoldenFileComparator implements GoldenFileComparator {
+/// environment.
+///
+/// See also:
+///
+///  * [GoldenFileComparator], the abstract class that
+///    [FlutterGoldenFileComparator] implements.
+///  * [FlutterGoldensRepositoryFileComparator], another
+///    [FlutterGoldenFileComparator] that tests golden images using the
+///    flutter/goldens repository.
+class FlutterSkiaGoldFileComparator extends FlutterGoldenFileComparator {
+  /// Creates a [FlutterSkiaGoldFileComparator] that will test golden file
+  /// images against Skia Gold.
+  ///
+  /// The [fs] and [platform] parameters useful in tests, where the default file
+  /// system and platform can be replaced by mock instances.
+  FlutterSkiaGoldFileComparator(
+    final Uri basedir,
+    this.skiaClient, {
+    FileSystem fs = const LocalFileSystem(),
+    Platform platform = const LocalPlatform(),
+  }) : super(
+    basedir,
+    fs: fs,
+    platform: platform,
+  );
+
+  final SkiaGoldClient skiaClient;
+
+  /// Creates a new [FlutterSkiaGoldFileComparator] that mirrors the relative
+  /// path resolution of the default [goldenFileComparator].
+  ///
+  /// The [goldens] and [defaultComparator] parameters are visible for testing
+  /// purposes only.
+  static Future<FlutterSkiaGoldFileComparator> fromDefaultComparator({
+    SkiaGoldClient goldens,
+    LocalFileComparator defaultComparator,
+  }) async {
+    defaultComparator ??= goldenFileComparator;
+    goldens ??= SkiaGoldClient();
+
+    final Directory baseDirectory = FlutterGoldenFileComparator.getBaseDirectory(goldens, defaultComparator);
+    if (!baseDirectory.existsSync())
+      baseDirectory.createSync(recursive: true);
+    await goldens.auth(baseDirectory);
+    await goldens.imgtestInit();
+    return FlutterSkiaGoldFileComparator(baseDirectory.uri, goldens);
+  }
+
   @override
-  Future<void> update(Uri golden, Uint8List imageBytes) async {
-    final File goldenFile = _getGoldenFile(golden);
-    await goldenFile.parent.create(recursive: true);
-    await goldenFile.writeAsBytes(imageBytes, flush: true);
+  Future<bool> compare(Uint8List imageBytes, Uri golden) async {
+    golden = _addPrefix(golden);
+    await update(golden, imageBytes);
+
+    final File goldenFile = getGoldenFile(golden);
+    if (!goldenFile.existsSync()) {
+      throw TestFailure('Could not be compared against non-existent file: "$golden"');
+    }
+    return await skiaClient.imgtestAdd(golden.path, goldenFile);
+  }
+
+  @override
+  Uri getTestUri(Uri key, int version) => key;
+
+  /// Decides based on the current environment whether goldens tests should be
+  /// performed against Skia Gold.
+  static bool isAvailableOnPlatform(Platform platform) {
+    final String cirrusCI = platform.environment['CIRRUS_CI'] ?? '';
+    final String cirrusPR = platform.environment['CIRRUS_PR'] ?? '';
+    final String cirrusBranch = platform.environment['CIRRUS_BRANCH'] ?? '';
+    final String goldServiceAccount = platform.environment['GOLD_SERVICE_ACCOUNT'] ?? '';
```

Comment:
Added the availability of the service account as a condition for using the FlutterSkiaGoldFileComparator. There are some shards that meet the first three conditions, but are not meant for executing skia gold testing.